addons/mail/static/src/js/services/mail_notification_manager.js

```diff
@@ -169,6 +169,22 @@ MailManager.include({
             }
         }
     },
+    /**
+     * Called when receiving a join and left channel notification:
+     *
+     * @private
+     * @param {Object} channelData
+     * @param {string} channelData.channel_partner_id
+     * @param {string} channelData.channel_partner_name
+     * @param {string} channelData.name
+     */
+    _handlePartnerJoinLeftNotification: function (channelData) {
+        var username = this.getSession().partner_id == channelData.channel_partner_id ?
```

Comment:
String concatenation on translatable string is a bad idea.

Here is a suggested way to make translatable string:

```
var title;
var content;
var isOnMyself = session.partner_id === channelData.channel_partner_id;
if (channelData.info === 'joined') {
    title = _t("Joined");
    if (isOnMyself) {
        content = _.str.sprintf(
            _t("You have joined channel <b>%s</b>."),
            channelData.name
        );
    } else {
        content = _.str.sprintf(
            _t("%s has joined channel <b>%s</b>."),
            channelData.channel_partner_name,
            channelData.name
        );
    }
} else if (channelData.info === 'left') {
    title = _t("Left");
    if (isOnMyself) {
        content = _.str.sprintf(
            _t("You have left channel <b>%s</b>."),
            channelData.name
        );
    } else {
        content = _.str.sprintf(
            _t("%s has left channel <b>%s</b>."),
            channelData.channel_partner_name,
            channelData.name
        );
    }
}
```